### ISSUE TITLE
Tagging ImageBundle makes it disappear

### ISSUE THREAD

**Issue by smokku** [bevyengine/bevy#1329]
- Bevy version: 0.4
- Operating system: Windows 10

I am building an UI and have an absolutely positioned image using `ImageBundle`. It works until I add an additional component to it (no matter empty or not) - image then disappears.

```rust
        .spawn(ImageBundle {
            style: Style {
                size: Size::new(Val::Px(25.0), Val::Px(25.0)),
                position_type: PositionType::Absolute,
                position: Rect {
                    left: Val::Px(10.0),
                    bottom: Val::Px(10.0),
                    ..Default::default()
                },
                ..Default::default()
            },
            material: materials.add(asset_server.load("crosshairs/crosshair026.png").into()),
            ..Default::default()
        })
        .with(control::MouseCursor)
```

The same code, but with `TextBundle` works as expected.

**Comment by Davier** [comment]
I was not able to reproduce using this code with current `master` branch:
```rust
use bevy::prelude::*;

fn main() {
    App::build()
        .add_plugins(DefaultPlugins)
        .add_startup_system(test.system())
        .run();
}

struct Test;

fn test(commands: &mut Commands, mut materials: ResMut<Assets<ColorMaterial>>, asset_server: Res<AssetServer>) {
    commands.spawn(CameraUiBundle::default());
    commands.spawn(ImageBundle {
            style: Style {
                size: Size::new(Val::Px(25.0), Val::Px(25.0)),
                position_type: PositionType::Absolute,
                position: Rect {
                    left: Val::Px(10.0),
                    bottom: Val::Px(10.0),
                    ..Default::default()
                },
                ..Default::default()
            },
            material: materials.add(asset_server.load("test.png").into()),
            ..Default::default()
        }).with(Test);
}
```
Could you give more details?

**Comment by smokku** [comment]
Updating to `master` didn't help.
So, I started stripping my app to the base minimum reproducible code.

This code:
```rust
use bevy::prelude::*;

fn main() {
    App::build()
        .add_plugins(DefaultPlugins)
        .add_startup_system(test.system())
        .run();
}

struct Test;

fn test(
    commands: &mut Commands,
    mut materials: ResMut<Assets<ColorMaterial>>,
    asset_server: Res<AssetServer>,
) {
    commands
        .spawn(CameraUiBundle::default())
        .spawn(NodeBundle {
            style: Style {
                size: Size::new(Val::Percent(100.0), Val::Percent(100.0)),
                position_type: PositionType::Absolute,
                justify_content: JustifyContent::Center,
                align_items: AlignItems::Center,
                ..Default::default()
            },
            material: materials.add(Color::NONE.into()),
            ..Default::default()
        })
        .with_children(|parent| {
            // bevy logo (image)
            parent.spawn(ImageBundle {
                style: Style {
                    size: Size::new(Val::Px(50.0), Val::Px(50.0)),
                    ..Default::default()
                },
                material: materials.add(asset_server.load("icon.png").into()),
                ..Default::default()
            });
        })
        .spawn(ImageBundle {
            style: Style {
                size: Size::new(Val::Px(50.0), Val::Px(50.0)),
                position_type: PositionType::Absolute,
                position: Rect {
                    left: Val::Px(110.0),
                    bottom: Val::Px(10.0),
                    ..Default::default()
                },
                ..Default::default()
            },
            material: materials.add(asset_server.load("icon.png").into()),
            ..Default::default()
        })
        // .with(Test)
        ;
}
```
renders:
![image](https://user-images.githubusercontent.com/147687/106029759-aeabf800-60cd-11eb-9ec5-b24e82d965d8.png)

After uncommenting `.with(Test)` I get:

![image](https://user-images.githubusercontent.com/147687/106029846-cd11f380-60cd-11eb-8842-de483bd303cb.png)

**Comment by DJMcNab** [comment]
I can reproduce. I'll try doing some debugging steps

**Comment by DJMcNab** [comment]
Yeah, so the bug is that the ImageBundle is closer than the near plane of the `CameraUIBundle`. Not sure how to fix it - I'm not an expert in the UI system.

**Comment by Davier** [comment]
The issue is that the `NodeBundle` somehow hides the second `ImageBundle`. Adding/removing the extra component would change the iteration order in some systems, and since both are on the same z plane the one drawn in front in unspecified. However `Color::None` is supposed to be transparent, and both icons show up when setting is_visible to false on the NodeBundle. So I believe there is a bug when drawing transparent materials that are in the same z plane. (something like #1211 could help in the future by making it impossible to happen).
I'm not familiar with the drawing/rendering part of bevy yet, so I'm not sure how to fix it.

**Comment by rparrett** [comment]
I can't reproduce this, since at least Bevy 0.6 at which point it's starting to get painful to test.

The last comment seems to be alluding to an underlying bug that I think should have been fixed by #5877 though.

Please reopen this if it turns out that's not the case.
